packages/framework/src/Model/Product/Brand/BrandFacade.php

```diff
@@ -110,19 +99,16 @@ public function create(BrandEditData $brandEditData)
 
     /**
      * @param $brandId
-     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\BrandEditData $brandEditData
+     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\BrandData $brandData
      * @return \Shopsys\FrameworkBundle\Model\Product\Brand\Brand
      */
-    public function edit($brandId, BrandEditData $brandEditData)
+    public function edit($brandId, BrandData $brandData)
     {
         $domains = $this->domain->getAll();
         $brand = $this->brandRepository->getById($brandId);
-        $brandData = $brandEditData->getBrandData();
         $brand->edit($brandData);
         $this->imageFacade->uploadImage($brand, $brandData->image->uploadedFiles, null);
-
-        $this->refreshBrandDomains($brand, $brandEditData);
-
+        $this->em->persist($brand);
```

Comment:
Entity cannot be responsible for persisting its secondary entities, because it doesn't have access to EntityManager. Entity is rather responsible just for creating the secondary entities, facade is responsible for persisting the created/updated entity so it starts the persistence cascade.

It's probably just semantics, I just want to be on the same page :)